File: lib/JFFUtils/NSArray/NSArray+NoThrowObjectAtIndex.m
#import "NSArray+NoThrowObjectAtIndex.h"

@implementation NSArray (NoThrowObjectAtIndex)

- (id)noThrowObjectAtIndex:(NSUInteger)index
{
    if ([self count] <= index) {
        return nil;
    }
    
    return self[index];
}

- (id)firstObject
{
    return [self noThrowObjectAtIndex:0];
}

@end

File: iAsync.Swift/lib/JUtils/JUtils/Array/NSArray+NoThrowObjectAtIndex.swift
//
//  NSArray+NoThrowObjectAtIndex.swift
//  JUtils
//
//  Created by Vladimir Gorbenko on 07.06.14.
//  Copyright (c) 2014 EmbeddedSources. All rights reserved.
//

import Foundation

public extension NSArray {
    
    //TODO remove
    func noThrowObjectAtIndex(index: Int) -> AnyObject? {
        
        if count <= index {
            return nil
        }
        
        return self[index]
    }
}
